### PR TITLE
EL-3115: Prepend MCC-exclusive endpoints with /mcc

### PR DESCRIPTION
**Pull request by aaj-moj**
https://dsdmoj.atlassian.net/browse/EL-3115

## What does this pull request do?

It takes all the code that's been added lately only for MCC use cases, and separates it so that it's easier to see what has been done exclusively for MCC.

More specifically changes these endpoints:
- `/cla_provider/api/v1/case/{{case_reference}}/detailed/` changed to `/cla_provider/api/v1/mcc/case/{{case_reference}}/detailed/`
- `/cla_provider/api/v1/case/{{case_reference}}/mcc_split/` changed to `/cla_provider/api/v1/mcc/case/{{case_reference}}/split/`

## Any other changes that would benefit highlighting?

While at it, the code related only to MCC has also been moved to a new package (django app) to keep this separation in code - not only in the URLs.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_backend/apps/mcc/tests/api/test_case_api.py
Comment (charlessco) [inline cla_backend/apps/mcc/tests/api/test_case_api.py:1]: Right now the tests focus on a successful response but don’t really confirm that the split behaved correctly or that the new case has the right data. It’d be worth strengthening the assertions and adding some negative or edge case scenarios to give better coverage.

- Consider things such as invalid payload
- Validating case details (create case with invalid details to show that only valid case works) 
- other edge cases not just happy path and errors

File: cla_backend/apps/mcc/tests/api/test_case_detailed_api.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def test_detailed_endpoint_case_ownership(self):
         response = self.client.get(url, HTTP_AUTHORIZATION=self.get_http_authorization())
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
-    @mock.patch("cla_provider.views.DetailedCaseSerializer")
+    @mock.patch("mcc.serializers.DetailedCaseSerializer")
```
Comment (charlessco) [inline cla_backend/apps/mcc/tests/api/test_case_detailed_api.py:129]: Maybe not mocking the serializes, creating a fake case or data would be a better approach shows that the code behaves how it should.

File: cla_backend/apps/mcc/tests/test_forms.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        child = split_form.save(user)
+        child.refresh_from_db()
+
+        self.case.refresh_from_db()
+        parent_mcc_form = SplitMCCCaseForm(case=self.case, request=self.request, data=split_data)
+        self.assertTrue(parent_mcc_form.is_valid(), msg="Parent MCC form blocked: %s" % parent_mcc_form.errors)
+        parent_mcc_form.save(user)
+
+        child_provider = child.provider
+        if child_provider is None:
```
Comment (charlessco) [inline cla_backend/apps/mcc/tests/test_forms.py:81]: if you going to have behavior logic, consider splitting the test.

File: cla_backend/apps/mcc/urls.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+from .views import MCCCaseViewSet
+
+case_split = MCCCaseViewSet.as_view({"post": "split"})
+case_detailed = MCCCaseViewSet.as_view({"get": "detailed"})
+
+urlpatterns = patterns(
+    "",
+    url(r"^case/(?P<reference>[A-Z|\d]{2}-\d{4}-\d{4})/split/$", case_split, name="case-split"),
+    url(r"^case/(?P<reference>[A-Z|\d]{2}-\d{4}-\d{4})/detailed/$", case_detailed, name="case-detailed"),
```
Comment (charlessco) [inline cla_backend/apps/mcc/urls.py]: Double check your regex, not sure its correct or could provide me an example url.

Comment (aaj-moj) [reply cla_backend/apps/mcc/urls.py]: Example URI: `/cla_provider/api/v1/mcc/case/YW-5432-7429/split/`

It is working for me... but if the case reference can have another form, would you prefer to see something more generic here?

Comment (charlessco) [reply cla_backend/apps/mcc/urls.py]: It works but introduces false positives. The stricter pattern `(?P<reference>[A-Z\d]{2}-\d{4}-\d{4})` performs better.

Comment (aaj-moj) [reply cla_backend/apps/mcc/urls.py]: Changed.

**Review by said-moj — APPROVED** [review]

**Review by charlessco — APPROVED** [review]
